### PR TITLE
Enable FindBugs on the verify stage

### PR DESCRIPTION
**Pull request by oleg-nenashev**
It also allows to setup a verification in pull request builders on https://jenkins.ci.cloudbees.com/job/plugins/job/maven-plugin/

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -49,6 +49,9 @@ THE SOFTWARE.
     <aetherVersion>0.9.0.M3</aetherVersion>
     <sisuInjectVersion>0.0.0.M5</sisuInjectVersion>
     <wagonVersion>2.4</wagonVersion>
+    <findbugs-maven-plugin.version>3.0.1</findbugs-maven-plugin.version>
+    <!--TODO: change to true after the code cleanup-->
+    <maven.findbugs.failure.strict>false</maven.findbugs.failure.strict>
```
Comment (KostyaSha) [inline pom.xml]: why in this plugin did you decide to set it with variable?

Comment (oleg-nenashev) [inline pom.xml]: I've improved the pattern a bit. I hope to streamline the implementation when and if we start implementing these features on the parent pom.

Comment (KostyaSha) [inline pom.xml]: Was it announced/discussed somewhere?

Comment (oleg-nenashev) [inline pom.xml]: Not yet. I'm planning to do it when I'm ready to start this activity

**Comment by oleg-nenashev** [comment]
@reviewbybees
There're 80 errors in FindBugs and about hundreds of errors in the aggressive Coverity working against the newest Jenkins core. I'm not going to cleanup this issues in the current iteration

File: pom.xml [hunk 2]
```diff
@@ -528,6 +531,24 @@ THE SOFTWARE.
           </systemPropertyVariables>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>findbugs-maven-plugin</artifactId>
+        <version>${findbugs-maven-plugin.version}</version>
```
Comment (stephenc) [inline pom.xml]: This is a redundant pom line

File: pom.xml [hunk 3]
```diff
@@ -528,6 +531,24 @@ THE SOFTWARE.
...
+        <configuration>
+          <failOnError>${findbugs.failOnError}</failOnError>
```
Comment (stephenc) [inline pom.xml]: this is a redundant pom line

**Comment by stephenc** [comment]
@oleg-nenashev you keep on adding completely unnecessary (or more correctly unjustified) pom bloat.

I wish you would stop doing that. However I will not block the review of this change because of your poor style of pom-craft

:bee:

**Comment by oleg-nenashev** [comment]
@stephenc 
It depends. At our previous company the POMs policy was to "keep all properties in POM if they may be altered in branches". For enforced policies like findbugs.failOnError it was also required to show them explicitly. POM files are supposed to be bloated till such configurations appear in Parent POMs.

File: src/main/java/hudson/maven/local_repo/PerExecutorLocalRepositoryLocator.java [hunk 4]
```diff
@@ -15,7 +16,13 @@ public PerExecutorLocalRepositoryLocator() {}
...
     @Override
     public FilePath locate(AbstractMavenBuild build) {
-        return build.getBuiltOn().getRootPath().child("maven-repositories/"+ Executor.currentExecutor().getNumber());
+        final Node builtOn = build.getBuiltOn();
+        final FilePath rootPath = builtOn != null ? builtOn.getRootPath() : null;
+        final Executor executor = Executor.currentExecutor();
+        if (rootPath == null || executor == null) {
+            return null;
+        }
+        return rootPath.child("maven-repositories/"+ executor.getNumber());
```
Comment (tfennelly) [inline src/main/java/hudson/maven/local_repo/PerExecutorLocalRepositoryLocator.java]: It looks like the above changes are just fixing an issue identified by Findbugs, right?

**Comment by tfennelly** [comment]
:bee:

File: src/main/java/hudson/maven/local_repo/PerExecutorLocalRepositoryLocator.java (hunk 4 above)
Comment (oleg-nenashev) [inline src/main/java/hudson/maven/local_repo/PerExecutorLocalRepositoryLocator.java]: Right. This code had too many errors in a single row, so I decided to fix it. BTW there're pretty much other issues

**Comment by stephenc** [comment]
@oleg-nenashev you are not working for that company now. The prefered style is to only express the absolute minimum required to produce a consistent build. The use of properties is generally an anti-pattern. 

There are some cases where it is valid to use properties, but there should be a justification for their use.

The rule of thumb should be: if you are expanding the property once and only once then you need an explicit justification provided for using a property.

**Comment by stephenc** [comment]
@oleg-nenashev FYI the Jenkins parent poms are over-bloated and make excessive incorrect use of properties that makes the proper use of Maven a lot harder than it needs to be.

**Pull request merged**